Feat/react 18

FIZLIN opened the pull request.

FIZLIN committed:
feat(react-18): WIP Updated some deps;

FIZLIN committed:
feat(react-18): Minor fix;

FIZLIN committed:
feat(react-18): Fixes;

FIZLIN committed:
feat(react-18): Udpated deps;

FIZLIN committed:
feat(react-18): Reverted react-unity-webgl version;

FIZLIN committed:
feat(react-18): Fixes;

FIZLIN committed:
feat(react-18): resolved routing issue 😮‍💨

FIZLIN committed:
feat(react-18): Updated types;

FIZLIN committed:
feat(react-18): Updated 'core' package versions;

FIZLIN committed:
feat(react-18): 'map3d' package versions;

FIZLIN committed:
feat(react-18): Fixes for 'ui-kit' package;

FIZLIN committed:
feat(react-18): 'ui-kit' lint 🧹

FIZLIN committed:
feat(react-18) 'resolved' ref issue

FIZLIN committed:
feat(react-18): Lowered dom types v.

FIZLIN committed:
feat(react-18): Resolved minor issue

FIZLIN committed:
feat(react-18): Resolved translation issues;

FIZLIN committed:
feat(react-18): Resolved translation issues;

FIZLIN committed:
feat(react-18): Naming tweaks;

FIZLIN committed:
feat(react-18): Rebase fix

FIZLIN committed:
feat(react-18): Fixes;

PAKovalev commented on packages/app/src/scenes/App.tsx:
```diff
@@ -99,6 +104,26 @@ const App: FC = () => {
+  //   );
+  // }
+
+  // // PUBLIC ROUTES
```
Why is it commented?

FIZLIN replied:
Can up when rebasing the onboarding changes. I've removed the comment.

dmitry-yudakov commented on packages/app/src/core/utils/router.utils.tsx:
```diff
@@ -1,31 +1,34 @@
 export const isTargetRoute = (currentPath: string, routes: RouteConfigInterface[]): boolean => {
   return routes.some((route) => {
-    return !!matchPath(currentPath, {path: route.path, exact: route.exact});
+    return !!matchPath({path: route.path}, currentPath);
```
Is it clear what replaces now missing `exact`? Or are we supposed to use something like `/a/b/*`? It doesn't feel right to just drop it.

dmitry-yudakov commented on packages/app/src/scenes/App.tsx:
```diff
@@ -49,23 +52,25 @@ const App: FC = () => {
-          theme={themeStore.theme}
-        />
-      </ThemeProvider>
+      <>
```
Is this extra `<></>` needed?

FIZLIN replied:
✅

FIZLIN committed:
feat(react-18): Resolve rebase issue + @types bump

FIZLIN committed:
feat(react-18): Fixes 😅

FIZLIN committed:
feat(react-18): Lint 🧹

FIZLIN committed:
feat(react-18): PropsWithChildren

FIZLIN committed:
feat(react-18): Lint;

dmitry-yudakov commented on packages/core/package.json:
```diff
@@ -15,11 +15,12 @@
+    "detect-browser": "^5.2.0",
+    "mobx-state-tree": "^5.1.3",
+    "react": "^18.2.0",
+    "react-com": "^1.0.6",
```
`react-com` seems like a mistake?

dmitry-yudakov commented on packages/core/package.json:
```diff
@@ -15,11 +15,12 @@
+    "mobx-state-tree": "^5.1.3",
+    "react": "^18.2.0",
+    "react-com": "^1.0.6",
+    "react-dom": "^16.14.0"
```
`react-dom` should be `^18.2.0`

FIZLIN replied:
✅

FIZLIN committed:
feat(react-18): Version fixes

FIZLIN committed:
feat(react-18): Lint 🧹

FIZLIN committed:
feat(react-18):Routing fix;

FIZLIN committed:
feat(react-18): Removed Strict Mode ⛔️
// We had to remove this because of the new React Strict Mode logic of rendering the component twice each time on dev
// This was causing problems with some of our Dialogs because with the useEffects being called twice, we were getting API issues.

FIZLIN committed:
feat(react-18):WIP Routing fix;

dmitry-yudakov committed:
fix(toast): hide built-in toastify icon

dmitry-yudakov committed:
fix(creator): broken default route

dmitry-yudakov committed:
feat(react-router): override useParams signature to return provided type

const { param } = useParam<{param: string}>();

react-router v6 returns param of type string|undefined - that has to be handled somehow.

We currently decided to put the responsibility of providing correct type to the user of useParams and enforce the type returned by it.

This it can be useParams<{param: string|undefined}>() if needed.

TODO remove param! assertion across the code.

FIZLIN committed:
Merge branch 'develop' of https://github.com/momentum-xyz/ui-client into feat/react-18

dmitry-yudakov committed:
Merge remote-tracking branch 'origin/develop' into feat/react-18

# Conflicts:
#	packages/app/src/scenes/odysseyCreator/pages/SpawnAssetPage/components/SpawnAssetMenu/SpawnAssetMenu.tsx

FIZLIN committed:
Merge branch 'develop' of https://github.com/momentum-xyz/ui-client into feat/react-18

FIZLIN committed:
feat(react-18): Lint;

FIZLIN committed:
feat(react): Resolved issues;

FIZLIN committed:
Merge branch 'develop' of https://github.com/momentum-xyz/ui-client into feat/react-18

FIZLIN committed:
feat(react-18): Protected route tweaks;
These had to be done becase there is no way to have something other than a `<Route(Fragment) />` as a child of `<Routes>` (prev. `<Switch />`)

FIZLIN committed:
feat(react-18): Routing tweaks;

FIZLIN committed:
feat(react-18): Resolved routing issue;

FIZLIN committed:
Merge branch 'develop' of https://github.com/momentum-xyz/ui-client into feat/react-18

kas0189-pa committed:
Merge branch 'develop' of https://github.com/momentum-xyz/ui-client into feature/react18_merge

 Conflicts:
	packages/app/src/scenes/unity/pages/UnityPage/UnityPage.tsx
	packages/app/src/scenes/widgets_OLD/pages/HelpWidget/components/Momentum/Momentum.tsx

PAKovalev committed:
Merge pull request #644 from momentum-xyz/feature/react18_merge

Feature/react18 merge

dmitry-yudakov committed:
fix(router): various issues with react-router v6

dmitry-yudakov committed:
Merge remote-tracking branch 'origin/develop' into feat/react-18

# Conflicts:
#	packages/ui-kit/src/atoms/Portal/Portal.ts

dmitry-yudakov committed:
feat(router): add some debug code for defaultRedirect logic

If we're in DEV env, show some info about the routes, rather than doing immediate redirect to default

dmitry-yudakov committed:
fix(build): typescript errors related to children prop

dmitry-yudakov committed:
fix(router): broken routing to nested creator paths

dmitry-yudakov committed:
fix(router): inability to click redirect link in debug env

dmitry-yudakov committed:
fix(router): treat exact flag as false by default in isTargetRoute

FIZLIN committed:
feat(react-18): Tab path matching;

FIZLIN committed:
Merge branches 'feat/react-18' and 'feat/react-18' of https://github.com/momentum-xyz/ui-client into feat/react-18

dmitry-yudakov committed:
feat(router): detect bad routing config, prevent redirecting loop

In case of redirecting to non-existing route we fall into busy loop that was hard to debug.

FIZLIN committed:
feat(react-18): Tweaks to tab logic;

FIZLIN committed:
Merge branch 'feat/react-18' of https://github.com/momentum-xyz/ui-client into feat/react-18
